### PR TITLE
Add Memory Inspector to Theia Repo

### PR DESCRIPTION
**Pull request by colin-grant-work**
#### What it does

This PR brings the code for inspecting memory on the frontend into the main Theia repo from the [Theia CPP Debug](https://github.com/eclipse-theia/theia-cpp-extensions) repository. Previously, that repository included a number of CPP-specific language features, but changes in Theia and in that repository have made it language-agnostic for the most part. At the same time, keeping it in a separate repo generates [its own problems](https://github.com/eclipse-theia/theia/issues/10042), so that deprecating that repository and moving the code here seems the best option.

#### How to test

1. Install the CDT-GDB VSCode plugin.
2. You should be able to run through the set of steps outlined [here](https://github.com/eclipse-theia/theia-cpp-extensions/pull/119), substituting `packages/memory-inspector/memory-inspector-example-workspace` for the `cpp-debug-workspace` of that repository.

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

Signed-off-by: Colin Grant <colin.grant@ericsson.com>

### CODE REVIEW

**Comment by colin-grant-work** [comment]
@paul-marechal, @vince-fugnitto, @marcdumais, we've discussed making this move eventually, and given that not having this code here caused an issue at the latest release, it seems like it may be time to get the machinery moving, at least.

File: packages/memory-inspector/memory-inspector-example-workspace/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+## Eclipse Theia - `@theia/cpp-debug` example workspace
```
Comment (marcdumais-work) [inline packages/memory-inspector/memory-inspector-example-workspace/README.md:1]: update?

File: packages/memory-inspector/memory-inspector-example-workspace/README.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+---
+
+### Description
+
+The workspace provides an easy and reproducible way to test the functionality present in the `@theia/memory-inspector` extension.
+
+note: ATM debugging does not work in Gitpod. Since GDB uses ptrace to attach to the debuggee and ptrace requires the docker container to be started with special permissions.
```
Comment (marcdumais-work) [inline packages/memory-inspector/memory-inspector-example-workspace/README.md:9]: I think this is no longer an issue, for a while.

Comment (marcdumais-work) [reply packages/memory-inspector/memory-inspector-example-workspace/README.md:9]: remove?

**Comment by marcdumais-work** [comment]
Nice. I think we should mention this at the next `dev-meeting`, so it doesn't look like we're trying to sneak this is. I do not expect push-back, but we might need to explain how this is now generic.

**Comment by colin-grant-work** [comment]
Closing for now - will reopen when I can remove the last vestiges of CDT-GDB peculiarity. Thanks @marcdumais, @vince-fugnitto, @paul-marechal

**Comment by marcdumais-work** [comment]
@colin-grant-work have we found another DA that could, potentially, work with this? Many/most languages do not care about dealing with raw memory, so the pool of candidates is probably limited.

**Pull request closed without merging**